feat(post): reinstate last updated date

LoneRifle opened the pull request:
## Problem

The last updated date in Post does not reflect the last time content
was updated, as Post.views is updated whenever someone views the post.
It was thus not returned by the backend, with the created timestamp being
not as useful and thus hidden

## Solution
- replace moment with date-fns-tz in /client
- avoid updatedAt changes on post view
- expose updatedAt to frontend
- use date-fns-tz to display the date after the answer section

## Screenshot
![image](https://user-images.githubusercontent.com/10572368/135206075-b25e2ce7-fe58-42c2-ab96-99df082a8bc5.png)

LoneRifle committed:
build(deps): remove moment in /client

The last updated date in Post does not reflect the last time content
was updated, as Post.views is updated whenever someone views the post.
Given this, remove references to moment for the time being, and the
moment package itself.

TODO: shard out views into its own table, taking after GoGovSG.

nauynix commented:
`UpdateAt` is being updated whenever someone views a post because it is incrementing the views count by one. However, it is possible to set the `silent` flag to prevent updating of updatedAt timestamp in a query.

https://sequelize.org/master/class/lib/model.js~Model.html#static-method-update

LoneRifle commented:
> `UpdateAt` is being updated whenever someone views a post because it is incrementing the views count by one. However, it is possible to set the `silent` flag to prevent updating of updatedAt timestamp in a query.
> 
> https://sequelize.org/master/class/lib/model.js~Model.html#static-method-update

good point! we still might have problems where viewers trying to change Post.views might be conflicting with other users trying to change the contents of the post, but I don't think this is a large problem for now

LoneRifle committed:
fix(post): avoid updatedAt changes on post view

LoneRifle committed:
feat(post): reinstate display of last updated

- expose updatedAt to frontend
- use date-fns-tz to display the date after the answer section

nauynix reviewed (approved):
lgtm
